authentication/src/main/java/io/camunda/authentication/config/WebSecurityConfig.java

```diff
@@ -242,11 +242,20 @@ public SecurityFilterChain unprotectedApiAuthSecurityFilterChain(
   public SecurityFilterChain protectedUnhandledPathsSecurityFilterChain(
       final HttpSecurity httpSecurity) throws Exception {
     // all resources not yet explicitly handled by any previous chain require an authenticated user
-    // thus by default unhandled paths are always protected
+    // thus by default unhandled paths will always return 404 not found
     return httpSecurity
         .securityMatcher("/**")
         .authorizeHttpRequests(
-            (authorizeHttpRequests) -> authorizeHttpRequests.anyRequest().denyAll())
+            authorizeHttpRequests -> authorizeHttpRequests.anyRequest().denyAll())
+        .exceptionHandling(
+            ex ->
+                ex.accessDeniedHandler(
+                    (request, response, accessDeniedException) ->
+                        response.sendError(HttpServletResponse.SC_NOT_FOUND)))
```

Comment:
💭  part 2
```suggestion
            // for unhandled paths return a 404 instead of a 403 - improves UX to detect misconfiguration of paths
            ex ->
                ex.accessDeniedHandler(
                    (request, response, accessDeniedException) ->
                        response.sendError(HttpServletResponse.SC_NOT_FOUND)))
```